flink-table/flink-table-runtime-blink/src/main/java/org/apache/flink/table/typeutils/BigDecimalTypeInfo.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.table.typeutils;
+
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.api.common.typeinfo.TypeInformation;
+import org.apache.flink.api.common.typeutils.base.BigDecComparator;
+import org.apache.flink.api.common.typeutils.base.BigDecSerializer;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+
+/**
+ * {@link TypeInformation} for {@link BigDecimal}.
+ *
+ * <p>It differs from {@link BasicTypeInfo#BIG_DEC_TYPE_INFO} in that:
+ * This type includes `precision` and `scale`, similar to SQL DECIMAL.
+ */
+public class BigDecimalTypeInfo extends BasicTypeInfo<BigDecimal> {
```

Comment:
Why not calling it `DecimalTypeInfo` ?

Reply:
I think it is a `TypeInfo` for `BigDecimal`, so `BigDecimalTypeInfo` is more appropriate.